### PR TITLE
Upgrade solidity to 0.8.11

### PR DESCRIPTION
**Pull request by wollooo**
and fix all the failing test that occured

### CODE REVIEW

File: packages/contracts/contracts/BorrowerOperations.sol [hunk 1]
```diff
@@ -12,10 +12,12 @@ import "./Dependencies/LiquityBase.sol";
 import "./Dependencies/Ownable.sol";
 import "./Dependencies/CheckContract.sol";
 import "./Dependencies/console.sol";
+import "./Dependencies/SafeMath.sol";
 
 contract BorrowerOperations is LiquityBase, Ownable, CheckContract, IBorrowerOperations {
+    using SafeMath for uint256;
     string constant public NAME = "BorrowerOperations";
-
+    bool public isInitialized;
```
Comment (udosson) [inline packages/contracts/contracts/BorrowerOperations.sol:20]: Could you set this as a comment, please?

File: packages/contracts/contracts/BorrowerOperations.sol [hunk 2]
```diff
@@ -124,6 +126,9 @@ contract BorrowerOperations is LiquityBase, Ownable, CheckContract, IBorrowerOpe
         checkContract(_sortedTrovesAddress);
         checkContract(_lusdTokenAddress);
         checkContract(_lqtyStakingAddress);
+        isInitialized = true;
```
Comment (udosson) [inline packages/contracts/contracts/BorrowerOperations.sol:129]: Could you set this as comment, please?

File: packages/contracts/contracts/CollSurplusPool.sol [hunk 3]
```diff
@@ -12,6 +14,7 @@ import "./Dependencies/console.sol";
 contract CollSurplusPool is Ownable, CheckContract, ICollSurplusPool {
     using SafeMath for uint256;
 
+    bool public isInitialized;
```
Comment (udosson) [inline packages/contracts/contracts/CollSurplusPool.sol:17]: Could you set this as a comment, please?

File: packages/contracts/contracts/DefaultPool.sol [hunk 4]
```diff
@@ -17,17 +19,18 @@ import "./Dependencies/console.sol";
  */
 contract DefaultPool is Ownable, CheckContract, IDefaultPool {
     using SafeMath for uint256;
-
+	bool public isInitialized;
```
Comment (udosson) [inline packages/contracts/contracts/DefaultPool.sol:22]: Could you set this as a comment, please?

File: packages/contracts/contracts/HintHelpers.sol [hunk 5]
```diff
@@ -84,12 +92,12 @@ contract HintHelpers is LiquityBase, Ownable, CheckContract {
...
 
         if (_maxIterations == 0) {
-            _maxIterations = uint(-1);
+            _maxIterations = type(uint256).max;
         }
 
         while (currentTroveuser != address(0) && remainingLUSD > 0 && _maxIterations-- > 0) {
             uint netLUSDDebt = _getNetDebt(troveManager.getTroveDebt(currentTroveuser))
-                .add(troveManager.getPendingLUSDDebtReward(currentTroveuser));
+                + (troveManager.getPendingLUSDDebtReward(currentTroveuser));
```
Comment (udosson) [inline packages/contracts/contracts/HintHelpers.sol:100]: Can your revert this back to `.add()`, please?

File: packages/contracts/contracts/LPRewards/Unipool.sol [hunk 6]
```diff
@@ -72,7 +74,9 @@ contract LPTokenWrapper is ILPTokenWrapper {
  * or first liquidity provider stakes UNIv2 LP tokens into it.
  */
 contract Unipool is LPTokenWrapper, Ownable, CheckContract, IUnipool {
+    using SafeMath for uint256;
     string constant public NAME = "Unipool";
+    bool public isInitialized;
```
Comment (udosson) [inline packages/contracts/contracts/LPRewards/Unipool.sol:79]: Could you set this as a comment, please?

File: packages/contracts/contracts/LQTY/LQTYStaking.sol [hunk 7]
```diff
@@ -15,6 +17,7 @@ import "../Interfaces/ILUSDToken.sol";
 contract LQTYStaking is ILQTYStaking, Ownable, CheckContract, BaseMath {
     using SafeMath for uint;
 
+    bool public isInitialized;
```
Comment (udosson) [inline packages/contracts/contracts/LQTY/LQTYStaking.sol:20]: Could you set this as a comment, please?

File: packages/contracts/contracts/LQTY/LockupContract.sol [hunk 8]
```diff
@@ -18,7 +18,7 @@ import "../Interfaces/ILQTYToken.sol";
 */
 contract LockupContract {
     using SafeMath for uint;
-
+	bool public isInitialized;
```
Comment (udosson) [inline packages/contracts/contracts/LQTY/LockupContract.sol:21]: Could you set this as a comment, please?

File: packages/contracts/contracts/LQTY/LockupContractFactory.sol [hunk 9]
```diff
@@ -26,6 +28,7 @@ import "../Dependencies/console.sol";
 contract LockupContractFactory is ILockupContractFactory, Ownable, CheckContract {
     using SafeMath for uint;
 
+	bool public isInitialized;
```
Comment (udosson) [inline packages/contracts/contracts/LQTY/LockupContractFactory.sol:31]: Could you set this as a comment, please?

File: packages/contracts/contracts/PriceFeed.sol [hunk 10]
```diff
@@ -23,6 +25,8 @@ import "./Dependencies/console.sol";
 contract PriceFeed is Ownable, CheckContract, BaseMath, IPriceFeed {
     using SafeMath for uint256;
 
+	bool public isInitialized;
```
Comment (udosson) [inline packages/contracts/contracts/PriceFeed.sol:28]: Could you set this as a comment, please?

File: packages/contracts/contracts/SortedTroves.sol [hunk 11]
```diff
@@ -46,12 +48,15 @@ import "./Dependencies/console.sol";
 contract SortedTroves is Ownable, CheckContract, ISortedTroves {
     using SafeMath for uint256;
 
+	bool public isInitialized;
```
Comment (udosson) [inline packages/contracts/contracts/SortedTroves.sol:51]: Could you set this as a comment, please?

File: packages/contracts/contracts/StabilityPool.sol [hunk 12]
```diff
@@ -147,6 +147,9 @@ import "./Dependencies/console.sol";
  */
 contract StabilityPool is LiquityBase, Ownable, CheckContract, IStabilityPool {
     using LiquitySafeMath128 for uint128;
+    using SafeMath for uint256;
+
+    bool public isInitialized;
```
Comment (udosson) [inline packages/contracts/contracts/StabilityPool.sol:152]: Could you set this as a comment, please?

**Review by udosson — CHANGES_REQUESTED** [review]
@wollooo thanks for the PR. Could you merge all commits into one commit after making the changes, please?

**Comment by wollooo** [comment]
All done, please see [#24](https://github.com/kumodao/borrowprot/pull/24)

**Comment by udosson** [comment]
@wollooo can this PR be closed?

**Pull request closed without merging**